### ISSUE TITLE
Ban calling specific autograd APIs from inside a function transform

### ISSUE THREAD

**Issue by zou3519** [pytorch/functorch#376]
These result in unexpected semantics.

Things to ban:
- requires_grad_
- retain_grad
- _fw_primal
- _make_dual
- _unpack_dual
- detach_()
- detach() (Maybe; shout if you're going to do this)

NB: We might still need to be able to call these functions inside the functorch framework (e.g. functorch.vjp, functorch.jvp)
